arch/riscv/compare256_rvv.c

```diff
@@ -0,0 +1,48 @@
+/* compare256_rvv.c - RVV version of compare256
+ * Copyright (C) 2023 SiFive, Inc. All rights reserved.
+ * Contributed by Alex Chiang <alex.chiang@sifive.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#if RISCV_RVV
+
+#include "../../zbuild.h"
+#include "fallback_builtins.h"
+
+#include <riscv_vector.h>
+
+static inline uint32_t compare256_rvv_static(const uint8_t *src0, const uint8_t *src1) {
+    uint32_t len = 0;
+
+    size_t vl;
+    long found;
+
+    do {
+        vl = __riscv_vsetvl_e8m4(256 - len);
```

Comment:
Why is it 256 - len? Do we need a comment for it?

Reply:
256 is the max length of data that we want to compare, `len` represents the length of the data we have processed, so the length of remaining part is `256 - len`.

Reply:
Does `v1` need to be reinitialized at the top of each loop? If the vector size is evenly divisible by 256 then it shouldn't correct?

Reply:
Perhaps it should just be `size_t v1 = vl = __riscv_vsetvl_e8m4(256);` at the top?

Reply:
In RISC-V, one way to handling large number of elements is "stripming" where each iteration of a loop handles some numbers of elemenets, and the iterations continue unitil all elemenets have been processed.

We have to update the `vl` value every time, which means how many elements we want to process in this iteration. We cannot set the `vl` in the top (outside of the for-loop), because `vl` may not be a factor of 256.

However, we indeed use a fixed `vl` under some conditions.

Reply:
Have you tested whether or not using a fixed value like `32` is faster than `256 - len`? Sorry to be a pain.

Reply:
If the vector register is large and allows users to process more than 32 elements in each iteration, using a fixed value can cause some performance waste.

Using `vl` in RISC-V RVV is a common way to partition a large problem size. However, sometimes we use a fixed `vl` for optimization, depending on our conditions.

Reply:
> We cannot set the vl in the top (outside of the for-loop), because vl may not be a factor of 256.

In what instances will `v1` not be a factor of 256? I believe, but could be wrong that vector sizes on other platforms are evenly divisible by 256.